src/process_utility.c

```diff
@@ -678,85 +685,145 @@ process_alter_column_type(Hypertable *ht, AlterTableCmd *cmd)
 }
 
 static void
-process_altertable(Node *parsetree)
+process_altertable_start(Node *parsetree)
 {
 	AlterTableStmt *stmt = (AlterTableStmt *) parsetree;
 	Oid			relid = AlterTableLookupRelation(stmt, NoLock);
-	Cache	   *hcache = NULL;
-	ListCell   *lc;
-	Hypertable *ht;
 
-	if (!OidIsValid(relid))
+	if (!OidIsValid(relid) || is_hypertable(relid))
 		return;
 
-	hcache = hypertable_cache_pin();
+	check_chunk_operation_allowed(relid);
+	process_altertable_plain_table(parsetree);
+}
 
-	/* TODO: forbid all alter_table on chunk table */
+static void
+process_altertable_subcmd(Hypertable *ht, Node *parsetree, ObjectAddress *obj)
```

Comment:
I think better to add _start or _end to this and next few functions to make it easier to follow the code.